sdk/dapp-kit/src/hooks/wallet/useAutoConnectWallet.ts

```diff
@@ -1,51 +1,62 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useEffect } from 'react';
+import { useQuery } from '@tanstack/react-query';
 
 import { useConnectWallet } from './useConnectWallet.js';
 import { useCurrentWallet } from './useCurrentWallet.js';
 import { useWallets } from './useWallets.js';
 import { useWalletStore } from './useWalletStore.js';
 
 export function useAutoConnectWallet(autoConnectEnabled: boolean) {
-	const { mutate: connectWallet } = useConnectWallet();
-	const setAutoConnectionStatus = useWalletStore((state) => state.setAutoConnectionStatus);
+	const { mutateAsync: connectWallet } = useConnectWallet();
 	const lastConnectedWalletName = useWalletStore((state) => state.lastConnectedWalletName);
 	const lastConnectedAccountAddress = useWalletStore((state) => state.lastConnectedAccountAddress);
 	const wallets = useWallets();
 	const { isDisconnected } = useCurrentWallet();
 
-	useEffect(() => {
-		if (
-			!autoConnectEnabled ||
-			!lastConnectedWalletName ||
-			!lastConnectedAccountAddress ||
-			!isDisconnected
-		) {
-			return;
-		}
+	useQuery({
```

Comment:
I'm not sure I see the benefit here between having this be a query with the sort of `useIsFetching` / `queryCache` code versus keeping it as a mutation with `useMutationState`. This query isn't cached and has no retries, so doesn't this effectively the same as what we previously had with the mutation in the `useEffect`?

I think I might just be confused about the vision here haha, but it feels "off" to have this be a query that is performing a side effect and returns data that can't be accessed by normal conventions

Reply:
The query is cached?